File: backend/src/routes/registry.ts
import express from 'express';
import path from 'path'
import {config} from '../setup.js'

const router = express.Router();

router.get('/registry/*', async (req, res) => {
  try {
    const url = path.join(config.registry, req.params[0]); // "https://registry.halzi.one:5000"
    const response = await fetch(url);
    const data = await response.json();

    res.json(data);
  } catch (error) {
    res.status(500).json({ error: 'Failed to fetch data' });
  }
});

export default router;

File: handler/registry.go
package handler

import (
	"app/config"

	"fmt"

	"github.com/gofiber/fiber/v2"

	"io"
	"log"
	"net/http"
)

func GetRegistryData(c *fiber.Ctx) error {
	url := fmt.Sprintf("%s/%s", config.Config("REGISTRY_HOST"), c.Params("*"))
	response, err := http.Get(url)
	if err != nil {
		log.Println(err)
		return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
			"error": "Failed to fetch data",
		})
	}
	defer response.Body.Close()
	data, err := io.ReadAll(response.Body)
	if err != nil {
		log.Println(err)
		return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
			"error": "Failed to fetch data",
		})
	}
	//fmt.Println(string(data))
	return c.Send(data)
}
